src/dialogs/more-info/controls/more-info-climate.js

```diff
@@ -309,11 +309,11 @@ class MoreInfoClimate extends EventsMixin(PolymerElement) {
   }
 
   supportsTemperature(stateObj) {
-    return (stateObj.attributes.supported_features & 1) !== 0;
+    return ((stateObj.attributes.supported_features & 1) !== 0) && stateObj.attributes.temperature;
   }
 
   supportsTemperatureRange(stateObj) {
-    return (stateObj.attributes.supported_features & 6) !== 0;
+    return ((stateObj.attributes.supported_features & 6) !== 0) && (stateObj.attributes.target_temp_low || stateObj.attributes.target_temp_high);
```

Comment:
test with `('target_temp_low' in stateObj.attributes)`

Reply:
Unfortunately this doesn't really work for the intended need:
<img width="744" alt="screen shot 2018-05-31 at 00 17 25" src="https://user-images.githubusercontent.com/7738048/40748148-23941496-6468-11e8-8dde-2b8bf8659a34.png">